### ISSUE TITLE
Reach

### ISSUE THREAD

**Issue by ynigoreyes** [ynigoreyes/TTUACM_Website#68]
Migrate to React since it is much easier to work with at scale. Migrate asap to mitigate technical debt

**Comment by ayushs08** [comment]
Hi,

I am currently working on migrating to React as well and so I've attached a document containing the research I did for the same.

The document attached covers 4 topics:
* Some overall things to be considered
* A little bit about backend
* SEO
* Performance

 Hope it helps. Cheers!

[Migrating to ReactJS.pdf](https://github.com/ynigoreyes/TTUACM_Website/files/2308880/Migrating.to.ReactJS.pdf)

Google docs:
https://docs.google.com/document/d/1lJgGKdxcE3kQjJS65Aql7J9Qr3XjTBDGdUbJXzoPDsU/edit?usp=sharing

**Comment by ynigoreyes** [comment]
Thanks for the help! @ayushs08 
One I would like to say about the article. ComponentWillMount lifecycle is deprecated and we should be using the constructor instead.

componentWillMount will only run the code when it mounts and will not listen to updates to the passed props. You also cannot access the props in componentWillMount.

**Comment by ynigoreyes** [comment]
Well the main difference is WHEN you use it.

The main difference is when they will be called. I believe that the constructor is called once no matter what. Throughout the whole UX, constructor is called once. componentDidMount is called every time the component is recreated. If you have a component that props in and out maybe like a modal, then componentDidMount will be called several times in the UX. 

I don't have a preference really. They are both just tools given to you to really express what you want the user to see @ayushs08

**Comment by ayushs08** [comment]
@ynigoreyes I completely agree with you. But I think I wasn't clear. What I was asking basically is whether using constructor explicitly or not makes a difference anymore?

As in is there a difference anymore between:

**THIS**

```
class Lorem extends React.Component {
  constructor(props) {
    super(props)
  }
}
```
and then using props as `this.props`

**AND** 

no constructor and just using props as `props.whatever`

**Comment by ynigoreyes** [comment]
Oh there's no difference really, you can access props either way.

I may lean more towards skipping the constructor to reduce boilerplate. No code is the best code 👍

**Comment by ayushs08** [comment]
Precisely ✌🏻

**Comment by ynigoreyes** [comment]
Because I hate CSS lol. And Material UI uses JSS. And I'm already using angular material.

**Comment by ayushs08** [comment]
Ah! Hahah makes sense. I myself am more inclined towards CSS but well to each their own 😊

**Comment by ynigoreyes** [comment]
Are you interested in contributing? I have a Google Calendar feature that I am working on, and after that feature we can start the migration.

**Comment by ayushs08** [comment]
Yes surely ✌🏻

**Comment by ynigoreyes** [comment]
Alright. Well I have not set up with project to be very team friendly. There is not a whole lot of documentation on it, especially on the front end so I will brief you as soon as i finish this feature

**Comment by ayushs08** [comment]
Sounds like a plan!
